### PR TITLE
Switch production bucket

### PR DESCRIPTION
**Pull request by ameukam**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Use a community-bucket for Kubernetes releases

#### Which issue(s) this PR fixes:

Part of:
 - https://github.com/kubernetes/release/issues/3729

#### Does this PR introduce a user-facing change?

```release-note
Use a community-owned bucket for releases
```

### CODE REVIEW

**Comment by ameukam** [comment]
cc @kubernetes/release-engineering 
/hold

File: pkg/release/release.go
Comment (cpanato) [inline pkg/release/release.go:107]: where can we check this value?

Comment (ameukam) [reply pkg/release/release.go:107]: @cpanato https://github.com/kubernetes/k8s.io/blob/5a417fe27eae5891761f8f636857fcb1e1baa173/infra/gcp/terraform/k8s-infra-releases-prod/main.tf#L38

**Review by saschagrunert — COMMENTED** [review]
@ameukam thank you for the PR, can we verify that it works using:

```
export TOOL_ORG=… && export TOOL_REF=… && go run ./cmd/krel stage
```

**Comment by ameukam** [comment]
> @ameukam thank you for the PR, can we verify that it works? For example using:
> 
> ```
> export TOOL_ORG=… && export TOOL_REF=… && go run ./cmd/krel stage
> ```

@saschagrunert https://console.cloud.google.com/cloud-build/builds;region=global/e0a3ad16-5f05-49e0-b867-3ba15712a5ed;tab=detail?project=kubernetes-release-test

**Review by saschagrunert — APPROVED** [review]

**Comment by ameukam** [comment]
FYI, I didn't find a way to validate this with the no-mock steps. I intend to use the release of `1.28.14` to validate this.

@saschagrunert Do we need to cut another release of the repo ?

File: pkg/release/release.go
Comment (saschagrunert) [reply pkg/release/release.go:107]: Is it possible for us to view the contents of the bucket using a short URL?

**Comment by saschagrunert** [comment]
> @saschagrunert Do we need to cut another release of the repo ?

Yeah we can do that after this PR lands.

File: pkg/release/release.go
Comment (ameukam) [reply pkg/release/release.go:107]: Yeah. I will have a PR against https://git.k8s.io/k8s.io that will allow release managers to view the content of the bucket.

**Review by cpanato — COMMENTED** [review]
thanks!

late lgtm

**Pull request merged**
